### PR TITLE
Modified submit page ui

### PR DESCRIPTION
**Pull request by dhanzy**
### What this PR does (required):
- Submt design for contests

### Screenshots / Videos (required):
- Screenshot (30)

### Any information needed to test this feature (required):
- npm install react-dropzone or npm install
- Sign in and Click on a contest
- You will be directed to the submit page for the contest

### Any issues with the current functionality (optional):
- Does not submit the image yet.

### CODE REVIEW

**Comment by HAZberi** [comment]
LGTM.

File: client/src/pages/SubmitPage/Submit.tsx [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+  filename: string;
+  status: string;
+  storageRef: string;
+  downloadUrl: string;
+  description: string;
+}
+
+export default function Submit(): JSX.Element {
+  const { loggedInUser } = useAuth();
+  if (loggedInUser === undefined) return <CircularProgress />;
```
Comment (rajivtitus) [inline client/src/pages/SubmitPage/Submit.tsx]: You can write this in a single if clause :)

File: client/src/pages/SubmitPage/Submit.tsx [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+  // On file upload (click the upload button)
+  const onFileUpload = () => {
+    if (imageList.length > 0) {
+      setProcessing(true);
+      imageList.forEach((image) => {
+        uploadImage({ image: image.file }).then((resp) => {
+          if (resp.success) {
+            addSubmissionToContest({ contestID: id, imageFile: resp.success }).then((resp) => {
+              if (resp.submission) {
+                console.log(resp.submission);
```
Comment (rajivtitus) [inline client/src/pages/SubmitPage/Submit.tsx]: Please remove and maybe update the snackbar here?

File: client/src/pages/SubmitPage/Submit.tsx [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+            setProcessing(false);
+            updateSnackBarMessage(resp.error ?? '');
+          }
+        });
+      });
+      return history.goBack();
+    }
+    updateSnackBarMessage('No image selected');
+  };
+  // const onFileUpload = () => {
```
Comment (rajivtitus) [inline client/src/pages/SubmitPage/Submit.tsx]: Please remove commented code if we no longer intend to use it

File: client/src/pages/SubmitPage/Submit.tsx [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+    <Container>
+      <Box my={10} width="70%" m="auto" className={classes.alignCenter}>
+        <Paper elevation={2}>
+          <Container className={classes.paperContainer}>
+            <Typography component="h2" variant="h3" className={classes.header}>
+              Submit Design
+            </Typography>
+            <ImagesDropZone setImageList={setImageList} />
+            <br />
+            {/* <input type="file" accept="image/*" onChange={onFileChange} /> */}
```
Comment (rajivtitus) [inline client/src/pages/SubmitPage/Submit.tsx]: Please remove

File: client/src/pages/SubmitPage/Submit.tsx [hunk 5]
```diff
@@ -0,0 +1,132 @@
...
+            <br />
+          </Container>
+        </Paper>
+        {imageList.length > 0 && <ImageElement imageList={imageList} handleDeleteImage={handleDeleteImage} />}
+        <Box className={classes.submitContainer}>
+          <button onClick={onFileUpload} className={classes.submitBtn}>
```
Comment (rajivtitus) [inline client/src/pages/SubmitPage/Submit.tsx]: Maybe use an MUI component here to stay consistent with app UI/design?

**Review by rajivtitus — APPROVED** [review]
Thanks for creating a new UI for image submission.

Just a few minor comments, please address before merging. Thanks!

File: client/src/pages/SubmitPage/ImageElement/ImageElement.tsx [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+
+function ImageElement({ imageList, handleDeleteImage }: ArrayImageList): JSX.Element {
+  const classes = useStyles();
+
+  return (
+    <Box my={2}>
+      <Paper>
+        <ImageList rowHeight={160} cols={4}>
+          {imageList.map((image, index) => (
+            <ImageListItem key={image.storageRef} style={{ maxHeight: '17.5em' }}>
```
Comment (rajivtitus) [inline client/src/pages/SubmitPage/ImageElement/ImageElement.tsx:29]: Avoid using indexes as keys, try to use something unique. When no unique keys are available, you can consider using a package like uuid :)

Comment (dhanzy) [reply client/src/pages/SubmitPage/ImageElement/ImageElement.tsx:29]: I did not use index as keys here. I used the generated image urls as key 
```
<ImageListItem key={image.storageRef} style={{ maxHeight: '17.5em' }}>
```
Where the key is image.storageRef

Comment (dhanzy) [reply client/src/pages/SubmitPage/ImageElement/ImageElement.tsx:29]: @rajivtitus

Comment (rajivtitus) [reply client/src/pages/SubmitPage/ImageElement/ImageElement.tsx:29]: My apologies, I was thinking about another PR I guess 😄

File: client/src/pages/SubmitPage/Submit.tsx (hunk 5 above)
Comment (dhanzy) [reply client/src/pages/SubmitPage/Submit.tsx]: Okay i will change that

File: client/src/pages/SubmitPage/ImageElement/ImageElement.tsx (hunk 6 above)
Comment (dhanzy) [reply client/src/pages/SubmitPage/ImageElement/ImageElement.tsx:29]: 😄 I understand.

**Comment by dhanzy** [comment]
Resolved all comment issues.

**Pull request merged**